rest/route/annotations.go

```diff
@@ -247,12 +247,26 @@ func (h *annotationByTaskPutHandler) Factory() gimlet.RouteHandler {
 func (h *annotationByTaskPutHandler) Parse(ctx context.Context, r *http.Request) error {
 	var err error
 	h.taskId = gimlet.GetVars(r)["task_id"]
+	taskExecutionsAsString := gimlet.GetVars(r)["execution"]
 	if h.taskId == "" {
 		return gimlet.ErrorResponse{
 			Message:    "task ID cannot be empty",
 			StatusCode: http.StatusBadRequest,
 		}
 	}
+	if taskExecutionsAsString == "" {
```

Comment:
I spoke to David about this, and he said that they only ever use the body of the request to set the execution and that reading the docs for the endpoint, wasn't even aware that they could send it as part of the url. We should instead of erring here, check if it's set in the body and only err if it's not set anywhere. We should also update the docs to say that the execution can be set in the url.

Reply:
Can we instead only allow it in the body and not in the URL?

Reply:
Or is this inconsistent with other routes ( / can we make them all consistent)

Reply:
Updated doc: https://github.com/evergreen-ci/evergreen/wiki/REST-V2-Usage#create-or-update-a-new-task-annotation

Reply:
It would definitely make it cleaner and less ambiguous to only take the execution in the body, but yes, taking the execution in the url seems to be the convention based on other routes.